app/src/main/java/io/homeassistant/companion/android/sensors/BluetoothSensorManager.kt

```diff
@@ -60,35 +76,67 @@ class BluetoothSensorManager : SensorManager {
             totalConnectedDevices = bluetoothDevices.filter { b -> b.connected }.count()
         }
         onSensorUpdated(
-            context,
-            bluetoothConnection,
-            totalConnectedDevices,
-            icon,
-            mapOf(
-                "connected_paired_devices" to connectedPairedDevices,
-                "connected_not_paired_devices" to connectedNotPairedDevices,
-                "paired_devices" to bondedString
-            )
+                context,
+                bluetoothConnection,
+                totalConnectedDevices,
+                icon,
+                mapOf(
+                        "connected_paired_devices" to connectedPairedDevices,
+                        "connected_not_paired_devices" to connectedNotPairedDevices,
+                        "paired_devices" to bondedString
+                )
         )
     }
 
+    private fun isBtOn(context: Context): Boolean {
+        var btOn = false
+        if (checkPermission(context, bluetoothState.id)) {
+            btOn = BluetoothUtils.isOn(context)
+        }
+        return btOn
+    }
+
     private fun updateBluetoothState(context: Context) {
         if (!isEnabled(context, bluetoothState.id))
             return
+        val icon = if (isBtOn(context)) "mdi:bluetooth" else "mdi:bluetooth-off"
+        onSensorUpdated(
+                context,
+                bluetoothState,
+                isBtOn(context),
+                icon,
+                mapOf()
+        )
+    }
 
-        var isBtOn = false
-
-        if (checkPermission(context, bluetoothState.id)) {
-            isBtOn = BluetoothUtils.isOn(context)
+    private fun updatedBLEDevice(context: Context): Boolean {
+        var result = false
+        var id1 = getSetting(context, bleTransmitter, BLE_ID1, "string", DEFAULT_BLE_ID1)
+        var id2 = getSetting(context, bleTransmitter, BLE_ID2, "string", DEFAULT_BLE_ID2)
+        var id3 = getSetting(context, bleTransmitter, BLE_ID3, "string", DEFAULT_BLE_ID3)
+        if (bleTransmitterDevice.id1 != id1 || bleTransmitterDevice.id2 != id2 || bleTransmitterDevice.id3 != id3) {
+            result = true
+            bleTransmitterDevice.id1 = id1
+            bleTransmitterDevice.id2 = id2
+            bleTransmitterDevice.id3 = id3
         }
-        val icon = if (isBtOn) "mdi:bluetooth" else "mdi:bluetooth-off"
+        return result
+    }
 
+    private fun updateBLEtrasnmitter(context: Context) {
+        if (!isEnabled(context, bleTransmitter.id) && bleTransmitterDevice.transmitting) // sensor has been turned off, stop transmitting
```

Comment:
After we stop what we need to stop we also need to `return` so none of the other logic gets executed since the sensor is disabled.

Reply:
Thanks, will fix this up.